src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java

```diff
@@ -146,7 +147,8 @@ public NeuralQueryBuilder(StreamInput in) throws IOException {
     @Override
     protected void doWriteTo(StreamOutput out) throws IOException {
         out.writeString(this.fieldName);
-        out.writeString(this.queryText);
+        out.writeOptionalString(this.queryText);
```

Comment:
Since either queryText and queryImage can be null we should use optional string here.